nipype/pipeline/engine/utils.py

```diff
@@ -486,7 +486,7 @@ def modify_paths(object, relative=True, basedir=None):
             out = tuple(out)
     else:
         if isdefined(object):
-            if isinstance(object, (str, bytes)) and os.path.isfile(object):
+            if isinstance(object, (str, bytes)) and os.path.isfile(os.path.join(basedir, object)):
```

Comment:
Please check whether this:

```suggestion
            if isinstance(object, (str, bytes)) and os.path.isfile(os.path.join(basedir, os.path.basename(object))):
```

Does not bring us back to the original problem. My impression is that it will. I agree with @effigies this is a result of loosing track of the traits.